src/jwtserviceauth.test.js

```diff
@@ -0,0 +1,311 @@
+            ).toISOString()
+          })
+        )
+      } else if (req.url === '/large_response') {
+        res.statusCode = 200
+        res.end('x'.repeat(1024))
+      } else if (req.url === '/echo') {
+        res.statusCode = 200
+        req.on('data', data => {
+          res.write(data)
+        })
+        req.on('end', () => {
+          res.end()
+        })
+      } else if (req.url === '/timeout') {
+        //
+      } else {
+        res.statusCode = 404
+        res.end()
+      }
+    } catch (e) {
+      res.statusCode = 400
+      res.end(JSON.stringify({ error: 'failed validation' }))
+    }
+  })
+
+  // Setup httpRequestHandler
+  let httpRequestHandlerR2 = null
+  let baseUrl = null
+  before(done => {
+    listenPromise.then(result => {
+      baseUrl = `http://localhost:${result.port}`
+      httpRequestHandlerR2 = (method, url, headers, body) => {
+        // Overwrite to point to test server
+        let parsedUrl = curl.parse(url)
+        url = `${baseUrl}${parsedUrl.path}`
+        // Do http request
+        let r2Request = r2[method.toLowerCase()](url, {
+          headers,
+          body
+        }).response
+        return r2Request.then(httpResponse => {
+          return httpResponse.arrayBuffer().then(data => {
+            return {
+              statusCode: httpResponse.status,
+              data: data,
+              headers: httpResponse.headers
+            }
+          })
+        })
+      }
+      console.log(`Listining on ${result.hostname}:${result.port}`)
+      done()
+    })
+  })
+
+  after(() => {
+    httpServer.close()
+  })
+
+  describe('httpRequest', () => {
+    it('should return 404', () => {
+      let response = httpRequest('GET', baseUrl)
+      return expect(response, 'to be fulfilled with value satisfying', {
+        statusCode: 404
+      })
+    })
+    it('should return too large', () => {
+      let response = httpRequest(
+        'GET',
+        `${baseUrl}/large_response`,
+        null,
+        null,
+        { maxResponseSize: 512 }
+      )
+      return expect(
+        response,
+        'to be rejected with error satisfying',
+        new HttpRequestError('Response too lange')
+      )
+    })
+    it('should return POST data', () => {
+      let response = httpRequest('POST', `${baseUrl}/echo`, null, 'Hello')
+      return expect(response, 'to be fulfilled with value satisfying', {
+        statusCode: 200
+      })
+    })
+    it('should timeout', () => {
+      let response = httpRequest('GET', `${baseUrl}/timeout`, null, null, {
+        timeout: 1
+      })
+      return expect(
+        response,
+        'to be rejected with error satisfying',
+        new HttpRequestError('Timeout')
+      )
+    })
+  })
+
+  describe('JwtServiceAuthError', () => {
+    it('innerError should be null', () => {
+      let error = new JwtServiceAuthError('')
+      expect(error.innerError, 'to be null')
+    })
+  })
+
+  describe('getGoogleAccessToken', () => {
```

Comment:
I would love it if we could split this into two different test files, so the logic and quirks of each provider is isolated?